tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -120,7 +117,8 @@ public void CloseSpan(Span span)
 
                 if (Iast.Iast.Instance.Settings.Enabled)
                 {
-                    IastRequestContext.AddsIastTagsToSpan(span, _iastRequestContext);
+                    _iastRequestContext?.AddIastVulnerabilitiesToSpan(span);
+                    OverheadController.Instance.ReleaseRequest();
```

Comment:
It looks like every root span (when IAST is enabled) will call `ReleaseRequest`, even if the original `AcquireRequest` returned false. Am I reading that correctly?

Reply:
That was a recent (and unfortunate) change that I did earlier to remove the releaseRequest call from the AddVulnerabilitiesToSpan, without adding the _iastRequestContext != null check. Thanks for spotting it.